615 integrating getuserinfo api

theboyofdream opened the pull request:
## Description
Integrated getUserInfo functionality on dashboard
Closes [615](https://github.com/TeamShiksha/openlogo/issues/615)

Functionality:
- updated `UserInfo.jsx` & `Usage.jsx` component to show user info fetched in dashboard & passed down to this components.
- updated `Home.jsx` to show `Go to Dashboard` when user is logged in otherwise show `Get Started` button.
- shows list of api keys from userdata to dashboard table.
- added `disabled` prop in CustomInput.jsx (because need for UserInfo card where we are passing disabled prop in email)
- redirecting user to `/dashboard` route after signin

Test cases:
- added new testcases for `UserInfo.jsx`, `Usage.jsx` & `Dashboard.jsx`
- updated signin testcase to cover redirect to `/dashboard` on signin
- wrapped all testcase renders of auth & signin with BrowserRouter

Extra:
- changed api endpoint in UserContext.jsx from `/api/user/data` to `/users/me`
- updated css class-names from camelCase to kebab-case on files which are directly related to dashboard & the component that renders in dashboard.
- removed few css clases which is not used in their respective component.

## What type of PR is this? (Check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📄 Documentation Update
- [x] 👨‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [x] ✅ Test
- [ ] 🛠️ CI/CD

## Screenshots (if applicable)

![Usage & UserInfo test coverage](https://github.com/user-attachments/assets/85ec109a-3b90-4086-a397-7093b20ee2b8)

![Dashboard & home test coverage](https://github.com/user-attachments/assets/7cfc36fc-e2f4-4369-99a2-6966da664765)

![Auth & Signin test coverage](https://github.com/user-attachments/assets/54b769a9-fab2-4d1f-bad6-04f377ae4fe5)

**User dashboard with API key**

![image](https://github.com/user-attachments/assets/2a61a839-4a69-415e-b672-417bd53e54e5)

**User dashboard with no API key**

![User with no API key](https://github.com/user-attachments/assets/02f853aa-1319-4d4c-94aa-af51b24ad70f)

**Demo video**

[Demo video](https://github.com/user-attachments/assets/3834cab7-7652-4286-b54f-d84d91153cd9)

## Checklist
- [x] I have performed a self-review of my code  
- [x] I have commented my code, particularly in hard-to-understand areas  
- [x] I have added tests that prove my fix is effective or that my feature works  
- [x] New and existing unit tests pass locally with my changes

theboyofdream committed:
getuserinfo api integration

- updated UserInfo.jsx, Usage.jsx component to render userdata passed from Dashboard.jsx
- updated api endpoint in UserContext.jsx from `/api/user/data` to `/users/me`
- added `disabled` prop in CustomInput.jsx (because need for UserInfo card where we are passing disabled prop in email)
- added test-cases for UserInfo.jsx, Usage.jsx & Dashboard.jsx
- updated css class-names from camelCase to kebab-case on files which are directly related to dashboard & the component that renders in dashboard.

theboyofdream committed:
added a new test-case for dashboard page to cover the table delete button render & clicks.

aps08 reviewed (changes requested):
user should be redirected to dashboard on login. How are you populating the API  KEY tables ? And sonar issue

theboyofdream commented:
> user should be redirected to dashboard on login. How are you populating the API KEY tables ? And sonar issue

- ok i will redirect them to dashboard
- for API KEY i using constants. Should i put dummy data on mongodb then use it?

aps08 commented:
> > user should be redirected to dashboard on login. How are you populating the API KEY tables ? And sonar issue
> 
> * ok i will redirect them to dashboard
> * for API KEY i using constants. Should i put dummy data on mongodb then use it?

What will happen if the user has not created any API_KEY ?

theboyofdream committed:
- added redirect to `/dashboard` route after signin
- updated signin to cover redirect to `/dashboard` on signin
- wrapped all testcase renders of auth & signin with BrowserRouter
- added api keys from userdata to dashboard table
- updated testcases of dashboard to cover api key table

theboyofdream commented:
> > > user should be redirected to dashboard on login. How are you populating the API KEY tables ? And sonar issue
> > 
> > 
> > 
> > * ok i will redirect them to dashboard
> > * for API KEY i using constants. Should i put dummy data on mongodb then use it?
> 
> What will happen if the user has not created any API_KEY ?

in new commit showing this as empty message `"Your api keys will be visible here, click on generate key to add new api key"` inside table

aps08 commented on packages/ui/__tests__/components/dashboard/usage/Usage.test.jsx:
```diff
@@ -0,0 +1,40 @@
+import { describe, expect, it } from "vitest";
+import Usage from "../../../../src/components/dashboard/usage/Usage";
+
+const mockUserData = {
```
mock should not be here check other test cases.

theboyofdream replied:
mockuserdata constant moved to constant file

aps08 commented on packages/ui/__tests__/components/dashboard/userinfo/UserInfo.test.jsx:
```diff
@@ -0,0 +1,66 @@
+import { describe, expect, it } from "vitest";
+import UserInfo from "../../../../src/components/dashboard/userinfo/UserInfo";
+
+const mockUserInfoProps = {
```
mock should not be here check other test cases.

theboyofdream replied:
yes using mockuserdata from constant file now

aps08 commented on packages/ui/__tests__/page/dashboard/Dashboard.test.jsx:
```diff
@@ -0,0 +1,117 @@
+import { BUTTON_TEXT } from "../../../src/utils/Constants";
+import { formatDate } from "../../../src/utils/Helpers";
+
+const mockUserData = {
```
mock should not be here check other test cases.

aps08 commented on packages/ui/__tests__/page/dashboard/Dashboard.test.jsx:
```diff
@@ -0,0 +1,117 @@
+      expect(screen.getByText("User Info")).toBeInTheDocument();
+      expect(screen.getByText("Change Password")).toBeInTheDocument();
+      expect(screen.getByText("Setting")).toBeInTheDocument();
+      // api key table headers
```
remove these kind of comments.

aps08 commented on packages/ui/__tests__/page/dashboard/Dashboard.test.jsx:
```diff
@@ -0,0 +1,117 @@
+    );
+
+    await waitFor(() => {
+      expect(screen.getByTestId("testid-dashboard")).toBeInTheDocument();
```
no need to check everything. Also use a constant to store the value and then use expect on the constant. Do not use screen.get directly inside expect statement.

aps08 commented on packages/ui/__tests__/page/dashboard/Dashboard.test.jsx:
```diff
@@ -0,0 +1,117 @@
+      expect(screen.getByText("Setting")).toBeInTheDocument();
+      // api key table headers
+      expect(screen.getByText("Description")).toBeInTheDocument();
+      expect(screen.getByText("Created")).toBeInTheDocument();
```
no need to check everything. Also use a constant to store the value and then use expect on the constant. Do not use screen.get directly inside expect statement.

aps08 commented on packages/ui/__tests__/page/dashboard/Dashboard.test.jsx:
```diff
@@ -0,0 +1,117 @@
+        const row = screen.getByText(key_description).closest("tr");
+        expect(within(row).getByText(key_description)).toBeInTheDocument();
+        expect(
+          within(row).getByText(formatDate(updated_at))
```
use constant

theboyofdream replied:
ok

aps08 commented on packages/ui/__tests__/page/dashboard/Dashboard.test.jsx:
```diff
@@ -0,0 +1,117 @@
+    );
+
+    await waitFor(() => {
+      expect(screen.getByTestId("testid-dashboard")).toBeInTheDocument();
```
no need to check everything. Also use a constant to store the value and then use expect on the constant. Do not use screen.get directly inside expect statement.

theboyofdream replied:
yes

aps08 committed:
Merge branch 'stage' into 615-integrating-getuserinfo-api

theboyofdream committed:
added apikey table headers to constant file
refactored testcases & dashboard page

theboyofdream committed:
Merge branch '615-integrating-getuserinfo-api' of https://github.com/TeamShiksha/openlogo into 615-integrating-getuserinfo-api

theboyofdream committed:
refactor: moved mockdata from testcases to constant files

theboyofdream committed:
added conditional button on hero-section & pricing card for logged in user
removed unused styles from changePassword component
added testcase for `Home.jsx`
updated other testcase which are using home component

theboyofdream committed:
Chore(ui): fixed sonarqube issue in `PricingCard.jsx`

theboyofdream commented:
> user should be redirected to dashboard on login. How are you populating the API KEY tables ? And sonar issue

done

theboyofdream commented:
> user should be redirected to dashboard on login. How are you populating the API KEY tables ? And sonar issue

added redirection to dashboard after successful login
Fixed sonar issue
